### PR TITLE
Model cleanup / refactoring

### PR DESCRIPTION
**Pull request by lechup**
* enable gis on models
* added gis fields
* cleanup of model structure

---
This change is [Reviewable](https://reviewable.io/reviews/enviromonitor/enviromonitorweb/25)

### CODE REVIEW

File: EnviroMonitorWeb/api/models.py [hunk 1]
```diff
@@ -1,29 +1,44 @@
...
+class Station(TimeTrackable):
 
     # Fields
     name = models.CharField(max_length=255)
-    created = models.DateTimeField(auto_now_add=True, editable=False)
-    last_updated = models.DateTimeField(auto_now=True, editable=False)
-    type = models.TextField(max_length=100)
-    notes = models.TextField(max_length=100)
+    # shouldn't be a choice field?
+    type = models.CharField(max_length=100)
```
Comment (lechup) [inline EnviroMonitorWeb/api/models.py]: @zen what do we want to put here? can You provide some help_text?

File: EnviroMonitorWeb/api/models.py [hunk 2]
```diff
@@ -1,29 +1,44 @@
...
+    notes = models.CharField(max_length=255)
+    # what is this field for?
     test = models.BooleanField()
```
Comment (lechup) [inline EnviroMonitorWeb/api/models.py]: @zen same here. What is test field for? Any help_text?

File: EnviroMonitorWeb/api/models.py [hunk 3]
```diff
@@ -38,22 +53,27 @@ def get_update_url(self):
         return reverse('api_station_update', args=(self.id,))
 
 
-class Metering(models.Model):
+class Measurment(models.Model):
```
Comment (lechup) [inline EnviroMonitorWeb/api/models.py]: @zen Metering vs Measurment ?

File: EnviroMonitorWeb/api/models.py [hunk 4]
```diff
@@ -38,22 +53,27 @@ def get_update_url(self):
...
 
     # Fields
-    created = models.DateTimeField(auto_now_add=True)
+    created = models.DateTimeField(auto_now_add=True, primary_key=True, db_index=True)
     pm01 = models.FloatField()
     pm25 = models.FloatField()
     pm10 = models.FloatField()
+    # what are those fields? why do we have 3 outs?
     temp_out1 = models.FloatField()
```
Comment (lechup) [inline EnviroMonitorWeb/api/models.py]: @zen why do we have 3 outs for temp and humidity?

File: EnviroMonitorWeb/api/models.py [hunk 5]
```diff
@@ -38,22 +53,27 @@ def get_update_url(self):
...
-    temp_out2 = models.FloatField(max_length=100)
+    temp_out2 = models.FloatField()
     temp_out3 = models.FloatField()
+    # what is int1 field?!
+    temp_int1 = models.FloatField()
```
Comment (lechup) [inline EnviroMonitorWeb/api/models.py]: @zen we have _int1 fields. What are those for? Any help_text?

File: EnviroMonitorWeb/api/models.py [hunk 6]
```diff
@@ -38,22 +53,27 @@ def get_update_url(self):
...
     hum_out1 = models.FloatField()
     hum_out2 = models.FloatField()
     hum_out3 = models.FloatField()
-    temp_int1 = models.FloatField()
-    hum_int1 = models.TextField(max_length=100)
+    # shouldn't be the same type like temp_int1?!
+    hum_int1 = models.CharField(max_length=100)
+    # https://en.wikipedia.org/wiki/Received_signal_strength_indication ?!
     rssi = models.FloatField()
```
Comment (lechup) [inline EnviroMonitorWeb/api/models.py]: Same, here. Any help_text for rssi? And rationale behind?

File: EnviroMonitorWeb/api/models.py [hunk 7]
```diff
@@ -38,22 +53,27 @@ def get_update_url(self):
...
+    # ?!
     bpress_out1 = models.FloatField()
```
Comment (lechup) [inline EnviroMonitorWeb/api/models.py]: I guess is button press, but stil kind of magic without any help_text ;)

File: EnviroMonitorWeb/api/models.py [hunk 8]
```diff
@@ -72,15 +92,31 @@ def get_update_url(self):
...
+    * job calculates avarage of readings from Measurment entries older than 2 weeks (mean
+    should be calulated per hour basis, one MeasurmentHistory entry per hour)
+    * entries that was used to calculate avarage should be cleaned from Measurment database
+    """
+    pass
+
+
+# do we actually need this one?
+# what is the rationale? grouping Stations? sharing? I would skip it for now...
 class Project(models.Model):
```
Comment (lechup) [inline EnviroMonitorWeb/api/models.py]: @zen can we factor out Project model? What is the rationale? Grouping Stations? Sharing? I would skip it for now... Is it necessary?

**Comment by zen** [comment]
Review status: 0 of 2 files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[EnviroMonitorWeb/api/models.py, line 22 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-22:-KaHGxtvSrzkZ7Vh11b2:b-8w6jjt) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L22)):*
<details><summary><i>Previously, lechup (Leszek Piątek) wrote…</i></summary><blockquote>

@zen what do we want to put here? can You provide some help_text? And shouldn't it be a choice field?
</blockquote></details>

Choice field or even FK. For now, let's make it choice: EM0.1, EM1.0, custom

---

*[EnviroMonitorWeb/api/models.py, line 25 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-25:-KaHHCmFMiivv4vR_lhi:bvlwe5f) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L25)):*
<details><summary><i>Previously, lechup (Leszek Piątek) wrote…</i></summary><blockquote>

@zen same here. What is test field for? Any help_text?
</blockquote></details>

Test=Yes indicates, that data sent from station should not be used and station is in "testing mode". This although generates another problem - when station is changed to Test=No, starting from this point of time data should be included in all averages, etc.

---

*[EnviroMonitorWeb/api/models.py, line 57 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-57:-KaHHt4LPxRIg_g2V6UN:b-ui9rd9) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L57)):*
<details><summary><i>Previously, lechup (Leszek Piątek) wrote…</i></summary><blockquote>

@zen Metering vs Measurment ?
</blockquote></details>

Metering might be more correct for scientific data.

---

*[EnviroMonitorWeb/api/models.py, line 64 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-64:-KaHI2ZJ1bIlFnm-QhzC:bgvj5sa) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L64)):*
<details><summary><i>Previously, lechup (Leszek Piątek) wrote…</i></summary><blockquote>

@zen why do we have 3 outs for temp and humidity?
</blockquote></details>

Because current station has 2 outside temperature sensors, and some of custom built stations have even more. One of the prototypes has 3 actually

---

*[EnviroMonitorWeb/api/models.py, line 68 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-68:-KaHIFwePoqJZuM6M0K9:btdqvfk) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L68)):*
<details><summary><i>Previously, lechup (Leszek Piątek) wrote…</i></summary><blockquote>

@zen we have _int1 fields. What are those for? Any help_text?
</blockquote></details>

Int / out - internal / external (temperature). Internal temperature is a temperature of air sucked by PM sensor

---

*[EnviroMonitorWeb/api/models.py, line 75 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-75:-KaHIeUgX6qLkGmkU3Kh:b-gzq9qm) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L75)):*
<details><summary><i>Previously, lechup (Leszek Piątek) wrote…</i></summary><blockquote>

Same, here. Any help_text for rssi? And rationale behind?
</blockquote></details>

It indicates RSSI of WiFi. It used to be useful for debugging "vanishing" stations

---

*[EnviroMonitorWeb/api/models.py, line 77 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-77:-KaHIq9XlDf16sowFapu:bk0kx7o) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L77)):*
<details><summary><i>Previously, lechup (Leszek Piątek) wrote…</i></summary><blockquote>

I guess is button press, but stil kind of magic without any help_text ;)
</blockquote></details>

No, it's barometric pressure ;)

---

*[EnviroMonitorWeb/api/models.py, line 110 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-110:-KaHIzr3F3un_JZBdzys:bvgmc06) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L110)):*
<details><summary><i>Previously, lechup (Leszek Piątek) wrote…</i></summary><blockquote>

@zen can we factor out Project model? What is the rationale? Grouping Stations? Sharing? I would skip it for now... Is it necessary?
</blockquote></details>

It's crucial, it allows people to run their anti-smog projects. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25)*

**Comment by lechup** [comment]
Review status: 0 of 2 files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[EnviroMonitorWeb/api/models.py, line 60 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHPhyl6UssBkvw6ILX:-KaHPhyl6UssBkvw6ILY:b-k69148) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L60)):*
> ```Python
>     # Fields
>     created = models.DateTimeField(auto_now_add=True, primary_key=True, db_index=True)
>     pm01 = models.FloatField()
> ```

What are units of pm's ug/m^3?

---

*[EnviroMonitorWeb/api/models.py, line 64 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-64:-KaHQ1lc2ig5JUuN_FXh:bnwcfhh) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L64)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

Because current station has 2 outside temperature sensors, and some of custom built stations have even more. One of the prototypes has 3 actually
</blockquote></details>

Hmm.. what about units? Celcius degrees? I guess all temperatures are in same units?

---

*[EnviroMonitorWeb/api/models.py, line 70 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHQVOa7ahhe572RH7X:-KaHQVOa7ahhe572RH7Y:bkpxa3k) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L70)):*
> ```Python
>     temp_int1 = models.FloatField()
>     hum_out1 = models.FloatField()
>     hum_out2 = models.FloatField()
> ```

Hm... What are units of humidity? Is it relative or absolute?

---

*[EnviroMonitorWeb/api/models.py, line 77 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-77:-KaHQLK822ccoKsZx3Gv:b-2axjwh) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L77)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

No, it's barometric pressure ;)
</blockquote></details>

Again, what about units? :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25)*

**Comment by zen** [comment]
Review status: 0 of 2 files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[EnviroMonitorWeb/api/models.py, line 64 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-64:-KaHT-9qiDU16nwcKnh6:b-o18ttc) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L64)):*
<details><summary><i>Previously, lechup (Leszek Piątek) wrote…</i></summary><blockquote>

Hmm.. what about units? Celcius degrees? I guess all temperatures are in same units?
</blockquote></details>

Yes, it's C

---

*[EnviroMonitorWeb/api/models.py, line 70 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHQVOa7ahhe572RH7X:-KaHSehY-DiOEvsPKyxQ:b-afq4ud) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L70)):*
<details><summary><i>Previously, lechup (Leszek Piątek) wrote…</i></summary><blockquote>

Hm... What are units of humidity? Is it relative or absolute?
</blockquote></details>

It's relative, all sensors measure relative humidity

---

*[EnviroMonitorWeb/api/models.py, line 77 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-77:-KaHTUm7FbucKvLckLDx:btdt2y9) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L77)):*
<details><summary><i>Previously, lechup (Leszek Piątek) wrote…</i></summary><blockquote>

Again, what about units? :)
</blockquote></details>

It's hPa - absolute pressure. We can then correct it to relative using station location and altitude from backend. I think altitude should be added to the model

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25)*

**Comment by lechup** [comment]
Review status: 0 of 2 files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[EnviroMonitorWeb/api/models.py, line 25 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-25:-KaHUaeqnEqBNM4InEcJ:beho7g9) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L25)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

Test=Yes indicates, that data sent from station should not be used and station is in "testing mode". This although generates another problem - when station is changed to Test=No, starting from this point of time data should be included in all averages, etc.
</blockquote></details>

Hm... So in testing mode we should:
* ignore all data sent by sensor - we do not show such sensor on map?
OR
* while setting test to False, cleanup all Metering entries that were POSTed by sensor with test=True?, we show sensor on map, and all data but we indicate that it is in test mode?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25)*

**Comment by zen** [comment]
Review status: 0 of 2 files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[EnviroMonitorWeb/api/models.py, line 25 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-25:-KaHXHmNSjDYkO_MYDWZ:bhehkt4) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L25)):*
<details><summary><i>Previously, lechup (Leszek Piątek) wrote…</i></summary><blockquote>

Hm... So in testing mode we should:
* ignore all data sent by sensor - we do not show such sensor on map?
OR
* while setting test to False, cleanup all Metering entries that were POSTed by sensor with test=True?, we show sensor on map, and all data but we indicate that it is in test mode?
</blockquote></details>

Sounds good, let's test it

---

*[EnviroMonitorWeb/api/models.py, line 60 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHPhyl6UssBkvw6ILX:-KaHYTKP_k5RsHO-InAG:bb77ufg) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L60)):*
<details><summary><i>Previously, lechup (Leszek Piątek) wrote…</i></summary><blockquote>

What are units of pm's ug/m^3?
</blockquote></details>

Yes.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25)*

**Comment by lechup** [comment]
Review status: 0 of 2 files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[EnviroMonitorWeb/api/models.py, line 25 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-25:-KaHhB5DtrlIheYJv2dU:bvd71t) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L25)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

Sounds good, let's test it
</blockquote></details>

But which option second one? There is OR in the middle ;)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25)*

**Comment by lechup** [comment]
Reviewed 10 of 10 files at r2.
Review status: 8 of 11 files reviewed at latest revision, 10 unresolved discussions.

---

*[EnviroMonitorWeb/api/models.py, line 22 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-22:-KaM54sI65Bu59u_9Ar9:b-896fix) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L22)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

Choice field or even FK. For now, let's make it choice: EM0.1, EM1.0, custom
</blockquote></details>

Done.

---

*[EnviroMonitorWeb/api/models.py, line 25 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-25:-KaM577pSMwKX8iJSZ-y:b-56selh) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L25)):*
<details><summary><i>Previously, lechup (Leszek Piątek) wrote…</i></summary><blockquote>

But which option second one? There is OR in the middle ;)
</blockquote></details>

Ok, this will be next PR.

---

*[EnviroMonitorWeb/api/models.py, line 57 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-57:-KaM5De8MtDBNMeh1y9s:b-896fix) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L57)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

Metering might be more correct for scientific data.
</blockquote></details>

Done.

---

*[EnviroMonitorWeb/api/models.py, line 60 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHPhyl6UssBkvw6ILX:-KaM5Eo5PfhbOwy8gJis:b-896fix) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L60)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

Yes.
</blockquote></details>

Done.

---

*[EnviroMonitorWeb/api/models.py, line 64 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-64:-KaM5Gzj5cYoKRwEIk21:b-896fix) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L64)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

Yes, it's C
</blockquote></details>

Done.

---

*[EnviroMonitorWeb/api/models.py, line 68 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-68:-KaM5HV9VdAdIeNHxo00:b-896fix) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L68)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

Int / out - internal / external (temperature). Internal temperature is a temperature of air sucked by PM sensor
</blockquote></details>

Done.

---

*[EnviroMonitorWeb/api/models.py, line 70 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHQVOa7ahhe572RH7X:-KaM5InmdTWzN78i811w:b-896fix) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L70)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

It's relative, all sensors measure relative humidity
</blockquote></details>

Done.

---

*[EnviroMonitorWeb/api/models.py, line 75 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-75:-KaM5K1xaPb7bK2bwiTf:b-896fix) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L75)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

It indicates RSSI of WiFi. It used to be useful for debugging "vanishing" stations
</blockquote></details>

Done.

---

*[EnviroMonitorWeb/api/models.py, line 77 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-77:-KaM5NNOUcQaH052gAUt:b-896fix) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L77)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

It's hPa - absolute pressure. We can then correct it to relative using station location and altitude from backend. I think altitude should be added to the model
</blockquote></details>

Done.

---

*[EnviroMonitorWeb/api/models.py, line 110 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaHEUbzCJsulS4KYSy7-r1-110:-KaM5Oz742p56uzGj5tM:b-896fix) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/c048dcb2b4bef4fd04bb9f61b65a5a172fd06520/EnviroMonitorWeb/api/models.py#L110)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

It's crucial, it allows people to run their anti-smog projects. 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25)*

**Comment by zen** [comment]
Review status: 8 of 11 files reviewed at latest revision, 12 unresolved discussions.

---

*[EnviroMonitorWeb/api/models.py, line 56 at r3](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaMLmDAQJFhxCPfljtT:-KaMLmDAQJFhxCPfljtU:b-pisxql) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/a7754dfea399eb3f56ec6928a4c380d860575fb4/EnviroMonitorWeb/api/models.py#L56)):*
> ```Python
>         null=True
>     )
>     temp_int1 = models.FloatField(
> ```

"temp_int_air1" might be more clear

---

*[EnviroMonitorWeb/api/models.py, line 74 at r3](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaMM-HzxyA6kPCR1FQX:-KaMM-I-DJGyCGLKa9Pm:b-r2ycwp) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/a7754dfea399eb3f56ec6928a4c380d860575fb4/EnviroMonitorWeb/api/models.py#L74)):*
> ```Python
>         null=True
>     )
>     hum_int1 = models.FloatField(
> ```

Similarly, "hum_int_air1" might be more clear

---

*[EnviroMonitorWeb/api/migrations/0002_auto_20170113_0845.py, line 183 at r3](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaMMIn4LxbvmUxfG4m0:-KaMMIn4LxbvmUxfG4m1:biq9fyd) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/a7754dfea399eb3f56ec6928a4c380d860575fb4/EnviroMonitorWeb/api/migrations/0002_auto_20170113_0845.py#L183)):*
> ```Python
>             model_name='station',
>             name='type',
>             field=models.CharField(choices=[('1', 'EM0'), ('2', 'EM1'), ('3', 'CUSTOM')], max_length=1),
> ```

EM0 -> EM0.1 (first version of station)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25)*

**Comment by lechup** [comment]
Reviewed 1 of 2 files at r1, 1 of 2 files at r3, 5 of 5 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[EnviroMonitorWeb/api/models.py, line 56 at r3](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaMLmDAQJFhxCPfljtT:-KaMVa0TiArXjNCmtFVw:b-896fix) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/a7754dfea399eb3f56ec6928a4c380d860575fb4/EnviroMonitorWeb/api/models.py#L56)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

"temp_int_air1" might be more clear
</blockquote></details>

Done.

---

*[EnviroMonitorWeb/api/models.py, line 74 at r3](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaMM-HzxyA6kPCR1FQX:-KaMVfAqsV5MxyH60Dku:b-896fix) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/a7754dfea399eb3f56ec6928a4c380d860575fb4/EnviroMonitorWeb/api/models.py#L74)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

Similarly, "hum_int_air1" might be more clear
</blockquote></details>

Done.

---

*[EnviroMonitorWeb/api/migrations/0002_auto_20170113_0845.py, line 183 at r3](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25#-KaMMIn4LxbvmUxfG4m0:-KaMVkMr19zHbbdYkZLQ:b-896fix) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/a7754dfea399eb3f56ec6928a4c380d860575fb4/EnviroMonitorWeb/api/migrations/0002_auto_20170113_0845.py#L183)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

EM0 -> EM0.1 (first version of station)
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25)*

**Comment by lechup** [comment]
Reviewed 1 of 2 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25)*

File: api/models.py [hunk 9]
```diff
@@ -38,53 +146,55 @@ def get_update_url(self):
...
-        return u'%s' % self.id
-
     def get_absolute_url(self):
-        return reverse('api_metering_detail', args=(self.id,))
+        return reverse('api_metering_detail', args=(self.created,))
 
-    def get_update_url(self):
-        return reverse('api_metering_update', args=(self.id,))
 
+class MeteringHistory(MeteringBase):
```
Comment (lechup) [inline api/models.py]: @zen @grillazz Now I'm considering changing this into `HistoryMetering` instead... What do You think?

**Comment by grillazz** [comment]
Reviewed 15 of 20 files at r5.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/25)*

**Comment by lechup** [comment]
I'll reopen it as a PR based on master branch...

**Pull request closed without merging**